test/powershell/Modules/ThreadJob/ThreadJob.Tests.ps1

```diff
@@ -0,0 +1,345 @@
+
+            (Get-Runspace).Count | Should Be ($rsStartCount + 2)
+
+        }
+        finally
+        {
+            Get-Job | where PSJobTypeName -eq "ThreadJob" | Remove-Job -Force
+        }
+
+        (Get-Runspace).Count | Should Be $rsStartCount
+    }
+
+    It 'ThreadJob jobs should work with Receive-Job -AutoRemoveJob' {
+
+        Get-Job | where PSJobTypeName -eq "ThreadJob" | Remove-Job -Force
+
+        $job1 = Start-ThreadJob -ScriptBlock { 1..2 | foreach { Start-Sleep -Seconds 1; "Output $_" } } -ThrottleLimit 2
+        $job2 = Start-ThreadJob -ScriptBlock { 1..2 | foreach { Start-Sleep -Seconds 1; "Output $_" } }
+        $job3 = Start-ThreadJob -ScriptBlock { 1..2 | foreach { Start-Sleep -Seconds 1; "Output $_" } }
+        $job4 = Start-ThreadJob -ScriptBlock { 1..2 | foreach { Start-Sleep -Seconds 1; "Output $_" } }
+
+        $null = $job1,$job2,$job3,$job4 | Receive-Job -Wait -AutoRemoveJob
+
+        (Get-Job | where PSJobTypeName -eq "ThreadJob").Count | Should Be 0
+    }
+}
+
+Describe 'Job2 Tests' -Tags 'CI' {
+
+    AfterEach {
+
+        Get-Job | where PSJobTypeName -eq "ThreadJob" | Remove-Job -Force
+    }
+
+    It 'Verifies StopJob API' {
+
+        $job = Start-ThreadJob -ScriptBlock { Start-Sleep -Seconds 60 } -ThrottleLimit 5
+        $job.StopJob($true, "No Reason")
+        $job.JobStateInfo.State | Should Be "Stopped"
+    }
+
+    It 'Verifies StopJobAsync API' {
+
+        $job = Start-ThreadJob -ScriptBlock { Start-Sleep -Seconds 60 } -ThrottleLimit 5
+        $job.StopJobAsync($true, "No Reason")
+        Wait-Job $job
+        $job.JobStateInfo.State | Should Be "Stopped"
+    }
+
+    It 'Verifies StartJobAsync API' {
+
+        $jobRunning = Start-ThreadJob -ScriptBlock { Start-Sleep -Seconds 60 } -ThrottleLimit 1
+        $jobNotRunning = Start-ThreadJob -ScriptBlock { Start-Sleep -Seconds 60 }
+
+        $jobNotRunning.JobStateInfo.State | Should Be "NotStarted"
+
+        # StartJobAsync starts jobs synchronously for ThreadJob jobs
+        $jobNotRunning.StartJobAsync()
+        $jobNotRunning.JobStateInfo.State | Should Be "Running"
+    }
+
+    It 'Verifies JobSourceAdapter Get-Jobs' {
+
+        $job = Start-ThreadJob -ScriptBlock { "Hello" } | Wait-Job
+
+        $getJob = Get-Job -InstanceId $job.InstanceId 2>$null
+        $getJob | Should Be $job
+
+        $getJob = Get-Job -Name $job.Name 2>$null
+        $getJob | Should Be $job
+
+        $getJob = Get-Job -Command ' "hello" ' 2>$null
+        $getJob | Should Be $job
+
+        $getJob = Get-Job -State $job.JobStateInfo.State 2>$null
+        $getJob | Should Be $job
+
+        $getJob = Get-Job -Id $job.Id 2>$null
+        $getJob | Should Be $job
+
+        # Get-Job -Filter is not supported
+        $result = Get-Job -Filter @{Id = ($job.Id)} 3>$null
+        $result | Should Be $null
+    }
+
+    It 'Verifies terminating job error' {
+
+        $job = Start-ThreadJob -ScriptBlock { throw "My Job Error!" } | Wait-Job
```

Comment:
Is this test same as https://github.com/PowerShell/PowerShell/pull/7169/files#diff-74d73c8bffa8cd97ca36a864c1dc71bbR172

Reply:
It is a slightly different test since it involves Receive-Job, so I'd like to keep it.